src/ontology/cl.Makefile

```diff
@@ -258,6 +258,36 @@ pattern_docs: $(ALL_PATTERN_FILES)
 		       -o ../../docs/patterns/
 
 
+# ----------------------------------------
+# CUSTOM OBO OUTPUT
+# ----------------------------------------
+OBO_EXPORT_OPTIONS = --merge-comments --strip-gci-axioms --strip-owl-axioms
+
+$(SUBSETDIR)/%.obo: $(SUBSETDIR)/%.owl | all_robot_plugins
+	$(ROBOT) uberon:obo-export --input $< $(OBO_EXPORT_OPTIONS) --obo-output $@
+
+$(ONT).obo: $(ONT).owl | all_robot_plugins
+	$(ROBOT) uberon:obo-export --input $< $(OBO_EXPORT_OPTIONS) --obo-output $@
```

Comment:
My only concerns with this approach is that we artificially break isomorphicity between owl and obo products (on content supported by OBO format). I am _ok_ with this suggestion, but I would ask @cmungall for a review, and consider the alternative, which is that `--merge-comments` is applied during preprocessing (was it called SRCMERGED?). Of course, I don't have anything against --strip-gci-axioms --strip-owl-axioms and in fact, I would kinda like that in ODK over `grep -v`.

Reply:
Makes me a bit queasy too, but round-tripping is already broken for OBO products  as untranslatable OWL axioms are stripped from header.  I'm not aware of any products that use the OBO GCI axioms @cmungall  - please yell if I'm wrong.  They are mainly an irritation as we get frequent tickets from people confused by them.

Reply:
> My only concerns with this approach is that we artificially break isomorphicity between owl and obo products (on content supported by OBO format).

As I said elsewhere: that horse has already left the barn. It left the barn at the latest when the ODK started stripping untranslatable axioms in all OBO products – a behaviour that is not configurable.

I personally couldn’t care less about that. From my point of view, the OBO format, for all its benefits, is a _legacy_ format and people who choose to use an OBO artefact over any other format should already  be aware that they may not get the full picture of what the ontology contains.

> but I would ask @cmungall for a review

You may notice that this is exactly what I did.

> consider the alternative, which is that --merge-comments is applied during preprocessing

Disagree. For two reasons.

First, merging the comments make them slightly less practical, because the annotations on those comments (cross-references) are merged as well, and there is no longer any way to know which cross-reference applies to which part of the comment. Okay, it may not be a big deal, but why should we make comments less useful in all artefacts in all formats just to accommodate the idiosyncrasies of one particular format?

Second, merging the comments early in the pipeline would not in fact be as easy as it sounds, because in this instance the comments are spread over two different files (comments from the editors in `cl-edit.owl`, auto-generated comments from the CellMark folks in the CLM-CL component). So we would need to merge the CLM-CL component first. OK, we could do that in preprocessing, but then we would need to leave that component out of `$(OTHER_SRC)` (basically stop treating as an ODK “component”, and more as a completely special file) -- otherwise it would be merged again later, along with its original comments.

And no, using `$(SRCMERGED)` is of no use, because `$(SRCMERGED)` is never actually used to build any release artefacts. It is used for some checks and for extracting the import seeds, nothing more.

I _did_ in fact try merging the comments early in the pipeline, and soon concluded that it was way too much of a hassle. Feel free to have a go at it if you want, but my position is that late merging specifically for the OBO products is the option that is the least disruptive to the existing pipelines.

Reply:
> As I said elsewhere: that horse has already left the barn. It left the barn at the latest when the ODK started stripping untranslatable axioms in all OBO products – a behaviour that is not configurable.

Note my emphasis on "isomorphic (on content supported by OBO format)". Deleting axioms that cant be rendered in the format is a bit different than merging.

I am thoroughly convinced by your argumentation, and I want to add that "rdfs:comment" is really not a thing we need to concern ourselves with too much - merge-comments is probably better than "drop random one", which is the alternative if the goal is "legal OBO".

 Just for future: I would argue strongly against merging _key properties_ like label and definition. Hopefully this is never going to be necessary!

Thanks for the explanations!

Reply:
> merge-comments is probably better than "drop random one", which is the alternative if the goal is "legal OBO".

For completeness, I have implemented the "drop random one" behaviour as an option to `obo-export`, should someone want that behaviour.

Basically with the next version of the Uberon plugin it will be possible to choose what happens when a class has more than one comment:

* merge them into a single one;
* discard them all;
* discard all but one (the one that is kept being chosen in a non-deterministic manner).

Reply:
> I would argue strongly against merging key properties like label and definition. Hopefully this is never going to be necessary!

I personally wouldn’t care (as I don’t care about the OBO format generally), but that’s noted.

But if we ever get to the point that there is a need to merge labels and/or definitions, then something must have gone horribly wrong long before the conversion to OBO, and it’d be much better to fix the root cause than trying to hide the dust under the rug during the OBO conversion.

Reply:
> For completeness, I have implemented the "drop random one" behaviour as an option to obo-export, should someone want that behaviour.

Soooo coool! Great. Eventually you will need to start thinking of the scope of your robot plugin, I think it goes way beyond Uberon already! But no worries, I can rename it to "gouttegd-tools" locally if I so wish it :D

> it’d be much better to fix the root cause

Exactly!

Reply:
> But no worries, I can rename it to "gouttegd-tools" locally if I so wish it

That is of course completely possible given the way the ROBOT plugins work, but that’s something that would be best avoided IMHO. If everybody starts renaming plugins locally according to their own wishes, it’s going to be a mess. I’d recommend that whenever a plugin is used, it is used under its original name, to minimise confusion.

The fact that a given plugin was initially named after a particular project because it was originally specifically intended for that project should not be a problem, and is not reason enough IMO to rename the plugin.

The [FlyBase plugin](https://github.com/FlyBase/flybase-robot-plugin) was initially intended solely for FlyBase’s ontologies (hence its name), and now CL is also using it. Is that reason enough to rename it? I don’t think so.

Reply:
Fine by me :)